app/hotels/src/map/HotelSnaps.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+
+import * as React from 'react';
+import { StyleSheet, Dimensions, View } from 'react-native';
+import Carousel from 'react-native-snap-carousel';
+import {
+  HotelSnap,
+  Address,
+  type OnDimensionsChange,
+} from '@kiwicom/react-native-app-common';
+
+type Props = {|
+  items: Object[], // todo
+  selectedIndex: number,
+  onSnapToItem: (index: number) => void,
+|};
+
+type State = {|
+  availableWidth: number,
+|};
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: '#fff',
+    height: 130,
+  },
+  slider: {
+    padding: 10,
+  },
+});
+
+const SNAP_WIDTH = 0.8;
+
+class HotelSnaps extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      availableWidth: Dimensions.get('screen').width,
+    };
+  }
+
+  componentDidMount = () => {
+    Dimensions.addEventListener('change', this.onDimensionsChanged);
+  };
+
+  componentWillUnmount = () => {
+    Dimensions.removeEventListener('change', this.onDimensionsChanged);
+  };
+
+  onDimensionsChanged = ({ screen: { width } }: OnDimensionsChange) => {
+    this.setState({ availableWidth: width });
+  };
+
+  renderItem = ({ item }: { item: Object, index: number }) => {
+    const { availableWidth } = this.state;
+    const width = availableWidth * SNAP_WIDTH;
+
+    return <HotelSnap width={width} hotel={item.data} />;
+  };
+
+  render = () => {
+    const { items, selectedIndex, onSnapToItem } = this.props;
+    const { availableWidth } = this.state;
+    const itemWidth = availableWidth * SNAP_WIDTH;
```

Comment:
Seems like this is calculated twice (render and renderItem). Try to be DRY... 🙂

Reply:
fixed